### PR TITLE
fix(android/engine): register lexical model in switchToNextKeyboard 🍒 

### PR DESCRIPTION
**Pull request by darcywong00**
🍒 pick of #5439 to stable-14.0

This fixes the following in switchToNextKeyboard
* Check for valid keyboard index
* register associated lexical model

Note: This contains temporary globe action `GlobeKeyAction.GLOBE_KEY_ACTION_SWITCH_TO_NEXT_KEYBOARD` which must be reverted before merging.

## User Testing
@keymanapp/testers

1. Load the PR build
2. Install the following keyboards: khmer_angkor, gff_amharic, sil_cameroon_qwerty
3. Wait for the keyboards to install and the associated dictionary for gff_amharic
4. In the Keyman app press the globe key several times (switches to next keyboard w/o KeyboardPicker)
    * Verify sil_euro_latin shows suggestions
    * Verify khmer_angkor does not show suggestions
    * Verify gff_amharic shows suggestions
    * Verify sil_cameroon_qwerty does not show suggestions

### CODE REVIEW

**Review by mcdurdin — COMMENTED** [review]
LGTM pending temp code removal and user testing pass.

Question, though, how would you test this without that temp code? Is it something that cannot be triggered without that change?

**Comment by darcywong00** [comment]
> Question, though, how would you test this without that temp code? Is it something that cannot be triggered without that change?

It would involve making an app that uses Keyman Engine for Android and 
1. assign KMManager.setGlobeKeyAction()
2. has a mix of keyboards and lexical models installed

**Comment by MakaraSok** [comment]
Tested on Emulator Pixel_2_API_29 (Android 10). All tests pass.

**Review by mcdurdin — APPROVED** [review]
LGTM. Please remove test code before merge

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.278](https://keyman.com/downloads/releases/stable/14.0.278)

**Pull request merged**
